c2rust-refactor/src/scripting/to_lua_ast_node.rs

```diff
@@ -295,12 +297,87 @@ impl UserData for LuaAstNode<P<Expr>> {
 // thread that did not acquire it.
 // @type TyAstNode
 unsafe impl Send for LuaAstNode<P<Ty>> {}
-impl UserData for LuaAstNode<P<Ty>> {}
+impl UserData for LuaAstNode<P<Ty>> {
+    fn add_methods<'lua, M: UserDataMethods<'lua, Self>>(methods: &mut M) {
+        methods.add_method("get_kind", |lua_ctx, this, ()| {
+            match &this.0.borrow().node {
```

Comment:
This should use the `AstName` trait instead of a match here if `TyKind` implements it (see `ast_manip/ast_names.rs`).